### DISCUSSION TITLE
Como criar novos projetos com a mesma estrutura usada em aula?

### DISCUSSION THREAD

**Discussion by oimarcella** [balta-io/forum#366] [Tira Dúvidas] [answered]
Quando crio um novo projeto com `dotnet new Console -o nomeprojeto`  meu projeto é criado com essa estrutura no Program.cs:
![image](https://github.com/balta-io/forum/assets/43733159/1c036618-70f8-4516-bbc5-102cd6aaa218)

E na aula, a estrutura é essa:
![image](https://github.com/balta-io/forum/assets/43733159/9ce2bcd6-6205-453e-b7fb-092eabba6742)

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @mlasa, como vai?

Basta usar a flag `--use-program-main` no fim do seu comando.

```
dotnet new console -o MEUPROJETO --use-program-main
```

**Comment by oimarcella** [comment]
Percebi também que a importação dos recursos fica implícita, fica mais difícil para quem está no começo para saber de onde vem os recursos pois fica "escondido".

Então no arquivo .csproj gosto de alterar ImplicitsUsing para disabled

```<ImplicitUsings>disabled</ImplicitUsings>```

Ele dá alguns erros porque não foi importado, mas é só importar, ou usar o quick fix colocando o mouse por cima e selecionar a opção.
Exemplo:

![image](https://github.com/balta-io/forum/assets/43733159/d16e4544-42c7-43c1-bb0b-acab9a5bbe1b)

meu .csproj
![image](https://github.com/balta-io/forum/assets/43733159/e9a3bfbe-65f0-44b2-9a98-025dc621bb91)

**Reply by andrebaltieri** [reply]
Bom dia, @mlasa, como vai?
Se preferir seguir igual do curso, basta instalar a versão do .NET que usamos lá... por exemplo: 3.1, 5, 6....
https://dotnet.microsoft.com/en-us/download/dotnet

Para criar um projeto com uma versão específica do .NET é só utilizar a flag `-f`:
```
dotnet new console -o MeuConsoleApp -f net3.1
dotnet new console -o MeuConsoleApp -f net5.0
dotnet new console -o MeuConsoleApp -f net6.0
...
```
